docs/notebooks/proton_radiography_particle_tracing_custom_source.ipynb

```diff
@@ -0,0 +1,524 @@
+    "    sim = prad.SyntheticProtonRadiograph(grid, source, detector, verbose=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, instead of using `create_particles` to create the particle distribution, we will use the `load_particles` function to use the particles we have created above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sim.load_particles(pos, vel, particle=particle)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After this point, the rest of the process for creating the radiograph is unchanged."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "nbsphinx-thumbnail": {
+     "tooltip": "Proton Radiography"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "sim.run()\n",
+    "\n",
+    "size = np.array([[-1, 1], [-1, 1]]) * 1.5 * u.cm\n",
+    "bins = [200, 200]\n",
+    "\n",
+    "hax, vax, intensity = sim.synthetic_radiograph(size=size, bins=bins)\n",
+    "\n",
+    "fig, ax = plt.subplots(figsize=(8, 8))\n",
+    "plot = ax.pcolormesh(\n",
+    "    hax.to(u.cm).value, vax.to(u.cm).value, intensity.T, cmap=\"Blues_r\",\n",
+    ")\n",
+    "cb = fig.colorbar(plot)\n",
+    "cb.ax.set_ylabel(\"Intensity\")\n",
+    "ax.set_aspect(\"equal\")\n",
+    "ax.set_xlabel(\"X (cm), Image plane\")\n",
+    "ax.set_ylabel(\"Z (cm), Image plane\")\n",
+    "ax.set_title(\"Synthetic Proton Radiograph\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Calling the `synthetic_radiograph` function with the `ignore_grid` keyword set will display the source profile (the radiograph calculated in the absence of any grid fields)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hax, vax, intensity = sim.synthetic_radiograph(size=size, bins=bins, ignore_grid=True)\n",
+    "\n",
+    "fig, ax = plt.subplots(figsize=(8, 8))\n",
+    "plot = ax.pcolormesh(\n",
+    "    hax.to(u.cm).value, vax.to(u.cm).value, intensity.T, cmap=\"Blues_r\",\n",
+    ")\n",
+    "cb = fig.colorbar(plot)\n",
+    "cb.ax.set_ylabel(\"Intensity\")\n",
+    "ax.set_aspect(\"equal\")\n",
+    "ax.set_xlabel(\"X (cm), Image plane\")\n",
+    "ax.set_ylabel(\"Z (cm), Image plane\")\n",
+    "ax.set_title(\"Source Profile\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig, ax = plt.subplots()\n",
+    "ax.plot(hax.to(u.cm).value, np.mean(intensity, axis=0))\n",
+    "ax.set_xlabel(\"X (cm), Image plane\")\n",
+    "ax.set_ylabel(\"Mean intensity\")\n",
+    "ax.set_title(\"Mean source profile\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  }
```

Comment:
Delete empty cell at the end of the notebook.  This is often missed because Jupyter notebook automatically adds the cell when you execute the last cell in the notebook.  It's just something we have to be aware of when committing new commits.